src/backend/storage/buffer/bufmgr.c

```diff
@@ -3351,15 +3351,9 @@ FlushRelationBuffers(Relation rel)
 	int			i;
 	BufferDesc *bufHdr;
 
-<<<<<<< HEAD
-	/* Open rel at the smgr level if not already done */
-	RelationOpenSmgr(rel);
-
 	if (!RelationUsesBufferManager(rel))
 		return;
```

Comment:
Are you sure that this `if` is still valid after deleting `RelationOpenSmgr`? At first glance, this condition checks the value of the `smgr_which` field, which is set in the `smgropen` function, which in turn was called in `RelationOpenSmgr`.

Reply:
see below and descrition